### PR TITLE
Make passing plugins as str objects a more obvious failure

### PR DESCRIPTION
**Pull request by nicoddemus**
See #855 

**CC** @xmo-odoo

### CODE REVIEW

File: _pytest/config.py [hunk 1]
```diff
@@ -80,6 +80,9 @@ def _prepareconfig(args=None, plugins=None):
     try:
         if plugins:
             for plugin in plugins:
+                if isinstance(plugin, str):
```
Comment (xmo-odoo) [inline _pytest/config.py]: Maybe warn in case of unicode strings as well? Not sure how pytest handles the 2/3 split though.

Comment (flub) [inline _pytest/config.py]: Using `py.builtin._basestring` IIRC

**Comment by flub** [comment]
I think it would be better to do this fix in pluggy rather then in py.test to be honest.

Also, is something like `not isinstance(plugin, (types.ModuleType, types.ClassType, types.InstanceType))` not more correct for this check?

**Comment by flub** [comment]
Hum, maybe the isinstance I suggest is not really in the spirit of ducktyping.  And hence the generic approach in pluggy not so great.  Furthermore a plugin could provide no hooks but only some fixtures in which case the generic approach in pluggy is also not so great.  So maybe my previous comment is wrong and this PR is the right approach anyway.

**Comment by xmo-odoo** [comment]
> Also, is something like not isinstance(plugin, (types.ModuleType, types.ClassType, types.InstanceType)) not more correct for this check?

Well you'd also need to add `object` to it as any arbitrary instance could be a plugin, but then you face the issue that `isinstance("foo", object)` is `True`.

> Furthermore a plugin could provide no hooks but only some fixtures in which case the generic approach in pluggy is also not so great.

It just happens that my original use case was indeed a plugin which only provided fixtures, no hooks. And my feature request was about excluding strings: I had thought that since -p takes a "string" designator for plugins it could also work for the plugins argument. It is an ID10T error, but one which I think has at least a bit of logic threaded through it.

File: _pytest/config.py (hunk 1 above)
Comment (nicoddemus) [inline _pytest/config.py]: Done, thanks! :smile:

**Comment by RonnyPfannschmidt** [comment]
random thought, could we just consider strings in the same manner we consider the -p option ?

**Comment by xmo-odoo** [comment]
That would certainly be an alternative.

**Comment by nicoddemus** [comment]
Good idea @RonnyPfannschmidt, updated the PR to interpret strings as module names instead. :smile:

**Comment by RonnyPfannschmidt** [comment]
+1

**Pull request merged**